sql/src/main/java/io/crate/analyze/ParameterContext.java

```diff
@@ -40,13 +41,25 @@
 
     private int currentIdx = 0;
 
-    public ParameterContext(Object[] parameters, Object[][] bulkParameters, @Nullable String defaultSchema) {
+    private final int headerFlags;
+
+    public ParameterContext(Object[] parameters, Object[][] bulkParameters,
+                            @Nullable String defaultSchema, Integer headerFlags) {
         this.parameters = parameters;
         this.defaultSchema = defaultSchema;
         if (bulkParameters.length > 0) {
             validateBulkParams(bulkParameters);
         }
         this.bulkParameters = bulkParameters;
+        if(headerFlags != null) {
```

Comment:
if you check for `null`, the constructor parameter muss be `@Nullable`
otherwise no need to do that